VisualizationBase/src/nodes/ViewItemNode.h

```diff
@@ -58,7 +62,13 @@ class VISUALIZATIONBASE_API ViewItemNode : public Super<UINode>
 		void setSpacingParent(ViewItemNode* spacingParent);
 		ViewItemNode* spacingParent() const;
 
+		virtual QJsonValue toJson() const;
+
 	private:
+		ViewItemNode(const ViewItem* parent);
+
+		const ViewItem* parent_{};
```

Comment:
It feels very very wrong for a node to have to know what visualization is displaying it. This is indicative of some problem in the design. At the moment you only use this `parent_` during `to/fromJson`. I can see two kinds of solutions:
- The node should contain all necessary information to convert itself to and from json. This means that instead of having `parent_` it should have the x/y positions you're talking about. Then, it's the item's reponsibility to update the node whenever these positions change.
- The second solution is more radical, and I'd like to get your opinion on it. Initially we created this split between viewNodes and viewItems, because the gridLayoutFormElement could only display nodes. Now that it can also display items directly, in theory we might not need node types any more. As this is a major change in the design that will have many ripple effects, I'm not saying we should do it now. I just want to know, what you think about it, and especially if you think this might cause any other problems.